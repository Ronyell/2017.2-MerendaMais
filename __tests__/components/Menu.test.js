import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';
import Menu from '../../src/components/Menu';
import MENU_ITENS from '../../src/constants/menuItens';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore();

const initialState = {
  counselor: {
    profile: {
      isPresident: true,
    },
  },
};

const store = mockStore(initialState);

describe('Testing Menu Side Bar', () => {
  it('Test Menu Rendering', () => {
    const wrapper = shallow(<Menu {...initialState} />, { context: { store } });
    expect(wrapper).toMatchSnapshot();
  });

  it('Test onPress searchSchool', () => {
    const wrapper = shallow(<Menu {...initialState} />, { context: { store } });
    const button = wrapper.findWhere(c => c.key() === 'searchSchool');
    expect(button.length).toEqual(1);
    button.simulate('press');
  });

  it('Test onPress profileInfoScreen', () => {
    const wrapper = shallow(<Menu {...initialState} />, { context: { store } });
    const button = wrapper.findWhere(c => c.key() === 'profileInfoScreen');
    expect(button.length).toEqual(1);
    button.simulate('press');
  });

  it('Test onPress schedulingVisit', () => {
    const wrapper = shallow(<Menu {...initialState} />, { context: { store } });
    const button = wrapper.findWhere(c => c.key() === 'schedulingVisit');
    expect(button.length).toEqual(1);
    button.simulate('press');
  });

  it('Test onPress seeLegislation', () => {
    const wrapper = shallow(<Menu {...initialState} />, { context: { store } });
    const button = wrapper.findWhere(c => c.key() === 'seeLegislation');
    expect(button.length).toEqual(1);
    button.simulate('press');
  });

  it('Test onPress scheduleMeeting', () => {
    const wrapper = shallow(<Menu {...initialState} />, { context: { store } });
    const button = wrapper.findWhere(c => c.key() === 'scheduleMeeting');
    expect(button.length).toEqual(1);
    button.simulate('press');
  });

  it('Test onPress notifications', () => {
    const wrapper = shallow(<Menu {...initialState} />, { context: { store } });
    const button = wrapper.findWhere(c => c.key() === 'notifications');
    expect(button.length).toEqual(1);
    button.simulate('press');
  });

  it('Test onPress doComplaint', () => {
    const wrapper = shallow(<Menu {...initialState} />, { context: { store } });
    const button = wrapper.findWhere(c => c.key() === 'doComplaint');
    expect(button.length).toEqual(1);
    button.simulate('press');
  });

  it('Test onPress logout', () => {
    const wrapper = shallow(<Menu {...initialState} />, { context: { store } });
    const button = wrapper.findWhere(c => c.key() === 'logout');
    expect(button.length).toEqual(1);
    button.simulate('press');
  });

  it('Test onPress manageRegisters', () => {
    const initialStateWithPresident = {
      counselor: {
        profile: {
          isPresident: true,
        },
      },
    };

    const wrapper = shallow(<Menu {...initialStateWithPresident} />, { context: { store } });
    const button = wrapper.findWhere(c => c.key() === 'manageRegisters');
    expect(button.length).toEqual(1);
    button.simulate('press');
  });
});
