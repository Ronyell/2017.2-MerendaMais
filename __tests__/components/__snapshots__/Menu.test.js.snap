// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Menu Side Bar Test Menu Rendering 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#FF9500",
      "borderLeftColor": "black",
      "borderLeftWidth": 1,
      "flex": 1,
      "height": 1334,
      "paddingTop": 20,
    }
  }
>
  <ScrollView>
    <MenuButton
      iconName="search"
      isLogout={false}
      key="searchSchool"
      onPress={[Function]}
      text="Pesquisar Escola"
    />
    <MenuButton
      iconName="account-box"
      isLogout={false}
      key="profileInfoScreen"
      onPress={[Function]}
      text="Perfil"
    />
    <MenuButton
      iconName="access-time"
      isLogout={false}
      key="schedulingVisit"
      onPress={[Function]}
      text="Agendar visita"
    />
    <MenuButton
      iconName="insert-invitation"
      isLogout={false}
      key="scheduleMeeting"
      onPress={[Function]}
      text="Agendar reunião"
    />
    <MenuButton
      iconName="notifications"
      isLogout={false}
      key="notifications"
      onPress={[Function]}
      text="Notificações"
    />
    <MenuButton
      iconName="gavel"
      isLogout={false}
      key="seeLegislation"
      onPress={[Function]}
      text="Consultar Legislação"
    />
    <MenuButton
      iconName="report-problem"
      isLogout={false}
      key="doComplaint"
      onPress={[Function]}
      text="Denunciar"
    />
    <MenuButton
      iconName="logout"
      isLogout={true}
      key="logout"
      onPress={[Function]}
      text="Sair"
    />
  </ScrollView>
</Component>
`;
